mesonbuild/dependencies/base.py

```diff
@@ -1450,10 +1450,12 @@ def _detect_dep(self, name: str, modules: T.List[T.Tuple[str, bool]], components
                     cfgs = [x for x in tgt.properties['IMPORTED_CONFIGURATIONS'] if x]
                     cfg = cfgs[0]
 
-                is_debug = self.env.coredata.get_builtin_option('buildtype') == 'debug'
                 if 'b_vscrt' in self.env.coredata.base_options:
+                    is_debug = self.env.coredata.get_builtin_option('buildtype') == 'debug'
                     if self.env.coredata.base_options['b_vscrt'].value in ('mdd', 'mtd'):
                         is_debug = True
```

Comment:
Still missing case `b_vscrt='none'` if I understand that correctly. Probably should fallback to `is_debug = self.env.coredata.get_builtin_option('debug')` in this case.

Reply:
When `b_vscrt` is `none`, then we just don't pass any CRT flags to the compiler. That means the compiler will use whatever is the default, which will usually be `/MD` or `/MT`, both of which are release CRTs.

On platforms where `b_vscrt` is not supported, it won't be in `coredata.base_options` at all, so this case is only hit when building with MSVC and friends.

Reply:
Basically, if `-Db_vscrt=none` and we try to follow the `debug` option with MSVC, the CMake dependency will use the debug CRT and Meson will use the release CRT and everything will blow up at link time or runtime.

Reply:
> That means the compiler will use whatever is the default

assuming `c(pp)_args` is empty.... never thought about `buildtype plain/custom` with completely custom build flags? 

> Meson will use the release CRT

as you stated: Meson will use nothing. It is in the responsibility of the user to supply the correct build flags within the environment in this case.

Reply:
buildtype `custom` does not mean "completely custom build flags". It means the user has set debug/optimized to values that [do not match the mapping in the buildtype table](https://mesonbuild.com/Builtin-options.html#build-type-options). If you can set `--debug` you can and should also set `-Db_vscrt` instead of passing in the CRT args using `$lang_args`.

With buildtype `plain`, the user may indeed set any $lang_args, and we have no idea what CRT they're going to use. Maybe we should look at `$lang_args` in that case.

In neither case we should look at the value of `debug`.